core/src/main/java/fr/sncf/osrd/speedcontroller/generators/LinearAllowanceGenerator.java

```diff
@@ -23,24 +21,30 @@ public LinearAllowanceGenerator(double begin, double end,
         this.value = allowanceValue;
     }
 
+    @SuppressWarnings("checkstyle:RightCurly")
     @Override
     public Set<SpeedController> generate(Simulation sim, TrainSchedule schedule,
                                          Set<SpeedController> maxSpeeds) {
         // find the percentage of the allowance to add to the whole path
         double percentage;
-        if (allowanceType.equals(MarginType.TIME))
+        if (allowanceType.equals(MarginType.PERCENTAGE))
             percentage = value;
-        else { // allowanceType = DISTANCE
+        else {
             var totalStopsDuration = schedule.getStopDuration();
             var expectedTime = getExpectedTimes(sim, schedule, maxSpeeds, TIME_STEP);
             var totalTime = expectedTime.lastEntry().getValue() - expectedTime.firstEntry().getValue();
-            var schemaLength = expectedTime.lastEntry().getKey() - expectedTime.firstEntry().getKey();
-            var n = schemaLength / 100000; // number of portions of 100km in the train journey
-            var totalAllowanceMinutes = n * value; // margin value is in minutes per 100km
-            var totalAllowanceSeconds = totalAllowanceMinutes * 60;
-            // the margin should not affect the stop durations,
-            // so the percentage is calculated without the total stop duration
-            percentage = 100.0 * totalAllowanceSeconds / (totalTime - totalStopsDuration);
+            if (allowanceType.equals(MarginType.DISTANCE)) {
+                var schemaLength = expectedTime.lastEntry().getKey() - expectedTime.firstEntry().getKey();
+                var n = schemaLength / 100000; // number of portions of 100km in the train journey
+                var totalAllowanceMinutes = n * value; // margin value is in minutes per 100km
+                var totalAllowanceSeconds = totalAllowanceMinutes * 60;
+                // the margin should not affect the stop durations,
+                // so the percentage is calculated without the total stop duration
+                percentage = 100.0 * totalAllowanceSeconds / (totalTime - totalStopsDuration);
+            }
+            else { // TIME
```

Comment:
```suggestion
            } else { // TIME
```

This removes the need for the SuppressWarning

Reply:
Ok